### ISSUE TITLE
FR: Define Lua Vars in ship and weapon tables

### ISSUE THREAD

**Issue by Kiloku** [scp-fs2open/fs2open.github.com#3699]
The initial idea came from @EatThePath. 
This would let ship and weapon tables to include arbitrary key-value pairs directly in their definition, such that a script could then fetch these values from any given ship or weapon object. 
This would scripts that are used to add special features and abilities to specific ship or weapon types.  

Example proposed syntax:
ships.tbl
```
$Name: Fancy Ship 1
$Luavar: int scripted_fancy_resource 200
$Luavar: bool can_use_taunt true

$Name Fancy Ship 2
$Luavar: int scripted_fancy_resource 500
$Luavar: bool can_use_taunt false
```

And then in some -sct.tbm:
```Lua
function FancyStuff:tryTaunt()
	local shipIter = mn.getShipList()
	for ship in shipIter do
		if ship.tabled_vars["can_use_taunt"] do  
			-- fancy forced target switching stuff here
		end
	end
end
```

At least 3 people (myself included) saw ways that their scripts could be improved with such a feature, bypassing the large effort needed to have setup steps that look for each object by class name and then applying the values.

**Comment by Goober5000** [comment]
We have an established method of configuring things on a per-ship basis though... use a config file.  The MediaVP scripts load per-ship-class information from text-based config files, and Axem's scripts load tons of information from JSON config files.

Bolting this functionality on to ships.tbl would be unwieldy from both a coding and a modding perspective.

**Comment by Kiloku** [comment]
After a lot of discussion in the #scp channel on Discord, we arrived at the idea of having `$Lua vars` be a section that'd contain a list of strings which should contain lua code like so:
```
$Lua vars:
ship.myvar='foo'
ship.nextvar=10
$end_lua_vars
```

Afterwards, we could call a lua-based parsing function built into the engine, that'd read each line and store it in a way that'd become accessible from a scripter's lua code.
```Lua
local shipInfo = {}
for name, sobj in pairs(shiplist) do
  local ship = {}
  for _, item in pairs(sobj.luavars) do
    assert(loadstring(item))()
  end
  shipInfo[name] = ship
end
```

**Comment by BMagnu** [comment]
Implemented in #3707
